src/main/java/com/thealgorithms/strings/ReverseWordInString.java

```diff
@@ -0,0 +1,41 @@
+package com.thealgorithms.strings;
+
+/**
+ Given an input string s, reverse the order of the words.
+ A word is defined as a sequence of non-space characters.
+ The words in s will be separated by at least one space.
+ Return a string of the words in reverse order concatenated by a single space.
+ Note that s may contain leading or trailing spaces or multiple spaces between two words.
+ The returned string should only have a single space separating the words.Do not include any extra spaces
+ */
+
+public class ReverseWordInString {
+
+    public static String reverseWords(String s) {
+
+        s = s.trim();
+        String res = "";
+        int start = 0;
+        int end;
+
+        while (start < s.length()) {
+            while (s.charAt(start) == ' ') {
+                start++;
+            }
+            end = start + 1;
+
+            while (end < s.length() && (s.charAt(end) != ' ')) {
+                end++;
+            }
```

Comment:
What do you think about moving this into a separate method like `findNextWord`?

Reply:
do you mean the complete method or on substring part? I am not sure about the idea need to give it more thought maybe because substring only extracts the next word

Reply:
The point is that the method `reverseWordsInString` is quite complex. The part market here _obviously_ computes the `start` and `end` for the next word in your input. You could introduce two methods:
```java
private static int findNextWord(final String inStr, final int curPos) {
    //...
}
```
and
```java
private static int computeThisWordLength(final inStr, final int curPos) {
    //...
}
```

Reply:
Thanks for the explanation but not quite clear how to break it above ways but I have done some simplification, please review.